index.js

```diff
@@ -40,7 +42,9 @@ const sendResponse = ( req, settings, fn ) => {
 	const {
 		isEnvelopeMode,
 		isRestAPI,
-		processResponseInEnvelopeMode
+		processResponseInEnvelopeMode,
+		expectStreamMode,
```

Comment:
Is the `expectStreamMode` param strictly necessary? I can imagine detecting the `Content-Type: application/x-ndjson` response header and starting streaming automatically. If you don't supply a `streamrecord` listener, it's not an issue: the streamed items will be ignored, and the response callback will be called with the final line, indistinguishable from a normal request.

Reply:
no, i suppose it isn’t. the only reason i can think of would be for some future endpoint that sends application/x-ndjson responses in some custom format (that isn’t 100-followed-by-200-or-500-envelopes), but until one exists, we should be able to remove this.

Reply:
hmm, on second thought, this param could also prevent bugs caused by our stream processing being implemented in the browser but not node. if you’re forced to explicitly indicate whether you want your application/x-ndjson responses to be unpacked in this convention, we can immediately and loudly fail on node, rather than silently returning the wrong result.

Reply:
done in d371828b7d78925f957214459577f9d5c5d936fa

Reply:
The `processResponseInStreamMode` flag causes some trouble in the Calypso data-layer patch (https://github.com/Automattic/wp-calypso/pull/47898#discussion_r538093979), and keeping it just to detect missing Node.js support might not be worth it.

What if we simply unpacked all `application/x-ndjson` responses automatically, and threw a "not supported" error in Node.js when seeing the `Content-Type: application/x-ndjson` header alone?

Then the `processResponseInStreamMode` flag can be removed and all the conditions that check it can be replaced with simply checking the `Content-Type` header.

Reply:
good point! done in 24f855f48bb41fcd1b4393efbd9d4abcb9b9e963